### PR TITLE
Algorithm pipelines edits

### PR DESCRIPTION
**Pull request by sunash**
Addresses some comments and discussions about the algorithm_pipelines PR.

### CODE REVIEW

File: acnportal/acnsim/interface.py [hunk 1]
```diff
@@ -471,32 +587,38 @@ def _convert_to_amp_periods(self, kwh, station_id):
...
-            network.magnitudes,
-            network.constraint_index,
-            network.station_ids,
+            infrastructure_info.constraint_matrix,
+            infrastructure_info.constraint_limits,
+            infrastructure_info.constraint_ids,
+            infrastructure_info.station_ids,
         )
 
     def is_feasible(
```
Comment (sunash) [inline acnportal/acnsim/interface.py:612]: Revert

File: acnportal/acnsim/tests/test_interface.py [hunk 2]
```diff
@@ -1,236 +1,387 @@
...
         self.assertTrue(self.interface.is_feasible({}))
 
-    def test_is_feasible_unequal_schedules(self):
+    def test_is_feasible_unequal_schedules(self) -> None:
         with self.assertRaises(InvalidScheduleError):
             self.interface.is_feasible(
                 {"PS-001": [1, 2], "PS-002": [3, 4, 5], "PS-003": [4, 5]}
             )
 
-    def test_is_feasible(self):
```
Comment (sunash) [inline acnportal/acnsim/tests/test_interface.py:218]: Revert

File: acnportal/algorithms/tests/testing_interface.py [hunk 3]
```diff
@@ -247,57 +250,39 @@ def is_feasible(
...
-            schedule_matrix, linear, violation_tolerance, relative_tolerance
-        )
-
-    def infrastructure_constraints_feasible(
-        self, rates, linear, violation_tolerance=1e-5, relative_tolerance=1e-7
-    ):
         infrastructure = self.infrastructure_info()
-        tol = np.maximum(
-            violation_tolerance, relative_tolerance * infrastructure.constraint_limits
+        return infrastructure_constraints_feasible(
```
Comment (sunash) [inline acnportal/algorithms/tests/testing_interface.py:254]: Look into numba for speeding up this function

**Comment by zach401** [comment]
LGTM. Discussed offline (online).

**Pull request merged**